python/lsst/afw/table/base/baseContinued.py

```diff
@@ -315,5 +315,15 @@ def __getattr__(self, name):
         except AttributeError:
             return getattr(self.columns, name)
 
+    def __str__(self):
+        if self.isContiguous():
+            return str(self.asAstropy())
+        else:
+            fields = ' '.join(x.field.getName() for x in self.schema)
```

Comment:
We discussed this offline, but I'll make a response here, after talking it over with @SimonKrughoff :

We don't want to be constantly patching over the annoyances we have with our non-contiguous tables. We should instead fix them to behave better in general. So, instead of adding extra special casing to make non-contiguity "nicer" in each individual case, we'll do a "minimum functionality" thing and fix the non-contiguous tables themselves, by e.g. making numpy views into them that behave correctly.